From ed4938038f7b1b2af5c7b47189e250e3cdbd5ac3 Mon Sep 17 00:00:00 2001
From: Tao Zeng <tao.zeng@amlogic.com>
Date: Tue, 11 Apr 2023 17:31:53 +0800
Subject: [PATCH] init: dump env/args if too many [2/2]

PD#SWPL-119265

Problem:
kernel will panic if too many boot args

Solution:
dump them for debug

Verify:
t7

Change-Id: Icc8597f10e71d761fb9186578901a7918f1e1ac4
Signed-off-by: Dezhen Wang <dezhen.wang@amlogic.com>
---
 init/main.c | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/init/main.c b/init/main.c
index 428a47a69e08..05dcf5e8ad3f 100644
--- a/init/main.c
+++ b/init/main.c
@@ -581,6 +581,19 @@ static int __init unknown_bootoption(char *param, char *val,
 		}
 		argv_init[i] = param;
 	}
+#ifdef CONFIG_AMLOGIC_ENV_DEBUG
+	if (panic_later) {
+		int k;
+
+		pr_err("Dump init args\n");
+		for (k = 0; argv_init[k] && k < MAX_INIT_ARGS; k++)
+			pr_err("[%2d]: %s\n", k, argv_init[k]);
+
+		pr_err("Dump env args\n");
+		for (k = 0; envp_init[k] && k < MAX_INIT_ENVS; k++)
+			pr_err("[%2d]: %s\n", k, envp_init[k]);
+	}
+#endif
 	return 0;
 }

--
2.25.1

